Objects/longobject.c

```diff
@@ -2903,8 +2904,6 @@ _PyLong_Frexp(PyLongObject *a, Py_ssize_t *e)
         shift_digits = (DBL_MANT_DIG + 2 - a_bits) / PyLong_SHIFT;
         shift_bits = (DBL_MANT_DIG + 2 - a_bits) % PyLong_SHIFT;
         x_size = 0;
```

Comment:
I think you want `x_size = shift_digits` here. Note that the block you've removed changes the value of `x_size`, and that `x_size` is used after the block.

Reply:
Thank you for catching this!